posts/marker-gene-smoothing/index.qmd

```diff
@@ -0,0 +1,714 @@
+                       ,reference_profiles = InSituType::ioprofiles
+                       ,n_clusts = 3
+                       )
+
+### clean up cell type names and save results back to seurat object
+clustdt <- data.table(clust_o = insitu$clust, row.names=names(insitu$clust))
+clustdt[,clust:=clust_o]
+clustdt[grep("T CD4", clust),clust:="T CD4"]
+clustdt[grep("T CD8", clust),clust:="T CD8"]
+clustdt[grep("B-cell", clust),clust:="B cell"]
+clustdt[,.N,by=.(clust)]
+sem@meta.data$clust <- NULL
+sem <- 
+Seurat::AddMetaData(sem
+                    ,data.frame(clustdt[,.(clust, clust_o)]
+                                ,row.names=names(insitu$clust)
+                                )
+                    )
+
+```
+
+Here we can plot the InSituType clusters ('clust') on the UMAP.
+This UMAP was created using 'Analytic Pearson residuals' (https://genomebiology.biomedcentral.com/articles/10.1186/s13059-021-02451-7 ) .
+
+```{r, umap_visualization_function, message=FALSE,eval=FALSE}
+#| code-fold: true
+
+umapf <- function(umapreduc
+                  ,clustercol
+                  ,semuse
+                  ,cls=NULL
+                  ,xlim = NULL
+                  ,ylim = NULL){
+  
+  umapd <-  
+    data.table(semuse@reductions[[umapreduc]]@cell.embeddings
+               ,keep.rownames = TRUE)
+  setnames(umapd, c(names(umapd)[2:3]), c("UMAP_1", "UMAP_2"))
+  obsmrk <- merge(data.table(semuse@meta.data), umapd
+                  ,by.x="cell_ID"
+                  , by.y="rn")
+  
+  obstxt <- obsmrk[,lapply(.SD, median),by=c(clustercol),.SDcols=paste0("UMAP_",1:2)]
+  
+    p <- 
+      ggplot(obsmrk, aes(UMAP_1, UMAP_2, color=.data[[clustercol]])) + 
+      geom_point(size=0.2) + 
+      theme_bw() + coord_fixed(xlim=xlim, ylim=ylim) + 
+      geom_label_repel(data=obstxt, aes(x=UMAP_1, y=UMAP_2, label=.data[[clustercol]]),show.legend=FALSE
+                       ,inherit.aes=FALSE,color='black')
+    
+  if(is.null(cls)){
+    p <- p +  
+      scale_color_manual(values=rep(unname(pals::alphabet()), 3)
+                         ,guide=guide_legend(override.aes=list(size=4)))
+    
+  } else {
+    p <- p +  
+      scale_color_manual(values=cls
+                         ,guide=guide_legend(override.aes=list(size=4)))
+  }
+  return(p)
+}
+
+```
+
+
+```{r eval=FALSE, message=FALSE}
+
+ctpal <- c('#C20088','#005C31','#2BCE48'
+           ,'#4C005C','#F0A0FF','#003380'
+           ,'#FFCC99','#8F7C00','#9DCC00'
+           ,'#191919','#94FFB5','#0075DC'
+           ,'#FFA8BB','#FFA405','#993F00'
+           ,'#808080')
+names(ctpal) <- c('endothelial','mDC','plasmablast'
+                  ,'b','B cell','pDC'
+                  ,'macrophage','a','mast'
+                  ,'T CD4','fibroblast','T CD8'
+                  ,'NK','Treg','neutrophil'
+                  ,'c')
+umapf("pearsonumap", "clust", sem, cls = ctpal)
+```
+
+```{r}
+#| eval: true
+#| echo: false
+
+knitr::include_graphics("./figures/umap_clust_all.png")
```

Comment:
It would be nice to have figure captions around these figures. Check other post to see the syntax.